numpy/_core/src/umath/stringdtype_ufuncs.cpp

```diff
@@ -1881,6 +1881,179 @@ zfill_strided_loop(PyArrayMethod_Context *context,
     return -1;
 }
 
+static NPY_CASTING
+string_partition_resolve_descriptors(
+        PyArrayMethodObject *self,
+        PyArray_DTypeMeta *NPY_UNUSED(dtypes[3]),
+        PyArray_Descr *given_descrs[3],
+        PyArray_Descr *loop_descrs[3],
+        npy_intp *NPY_UNUSED(view_offset))
+{
+    if (given_descrs[3] || given_descrs[4] || given_descrs[5]) {
```

Comment:
More out of curiosity than anything else, but why doesn't `out` work? Isn't it fine as long as the dtypes are consistent?

Reply:
Right now all the `np.strings` wrappers don’t accept `out`. They could, in the future, but I figured it wasn’t worth going out of my way to add code paths for in-place operations that can’t happen yet. The fixed-width versions heavily leverage `out` by precomputing the output width, but they don’t support user-supplied `out` unless someone calls the ufunc directly with the private name.